src/main/java/it/gov/pagopa/fdr/to/eventhub/util/CommonUtil.java

```diff
@@ -251,4 +265,64 @@ private boolean sendEventToHub(
       return false;
     }
   }
+
+
+  /**
+   * Send a message to the Event Hub
+   */
+  private boolean sendEventBatchToHub(
+      List<String> jsonPayloads,
+      EventHubProducerClient eventHubClient,
+      FlussoRendicontazione flusso,
+      ExecutionContext context) {
+
+    try {
+
+      //
+      EventDataBatch evhEventBatch = eventHubClient.createBatch();
+      int batchMaxSize = evhEventBatch.getMaxSizeInBytes();
+      context.getLogger()
+          .fine(() ->
+              String.format("Defining batches with maximum dimension of [%s] bytes.",
+                  batchMaxSize));
+
+      for (String jsonPayload : jsonPayloads) {
+
+        EventData eventData = new EventData(jsonPayload);
+        eventData
+            .getProperties()
+            .put(SERVICE_IDENTIFIER, flusso.getMetadata().getOrDefault(SERVICE_IDENTIFIER, "NA"));
+
+        // Try to add the event from the array to the batch
+        if (!evhEventBatch.tryAdd(eventData)) {
+
+          // If the batch is full, send it and then create a new batch
+          eventHubClient.send(evhEventBatch);
+          evhEventBatch = eventHubClient.createBatch();
+
+          // Try to add that event that couldn't fit before.
+          if (!evhEventBatch.tryAdd(eventData)) {
```

Comment:
this is the case in which (`eventData` size) > (`evhEventBatch` size), it's not feasible, but I would test it

Reply:
Technically, this is a "dead branch", never reached with the current data structures of `CtDatiSingoloPagamento`, but it is added as "parachute" branch. Maybe this could not be tested via `Fdr1->fdr-to-evh` flow but it could be tested storing a XML zipped file with very large IUV/IUR field and automatically triggering the function. I will try it and, if possible, I will include here the results.